analyzers/src/SonarAnalyzer.VisualBasic/Rules/WeakSslTlsProtocols.cs

```diff
@@ -34,11 +34,34 @@ public sealed class WeakSslTlsProtocols : WeakSslTlsProtocolsBase<SyntaxKind, Id
     {
         protected override GeneratedCodeRecognizer GeneratedCodeRecognizer { get; } = VisualBasicGeneratedCodeRecognizer.Instance;
 
-        protected override SyntaxKind SyntaxKind { get; } = SyntaxKind.IdentifierName;
+        protected override SyntaxKind SyntaxKindOfInterest { get; } = SyntaxKind.IdentifierName;
 
         protected override DiagnosticDescriptor Rule { get; } = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
         protected override string GetIdentifierText(IdentifierNameSyntax identifierNameSyntax) =>
             identifierNameSyntax.Identifier.Text;
+
+        protected override bool IsNodeOfInterest(SyntaxNode node)
+        {
+            if (!(node.Parent is MemberAccessExpressionSyntax))
+            {
+                return false;
+            }
+
+            var current = node;
+            var parent = current.Parent;
+            while (parent != null && !parent.IsKind(SyntaxKind.IfStatement))
```

Comment:
I've realized now that the condition ca be in different constructs too (e.g. Ternary operator). I suggest to document the FP and let it be for the moment. We could handle it next time we touch this rule.